File: src/Klemm.One/Klemm.One.App/Helper/Interop/LocalStorage.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.JSInterop;

namespace Klemm.One.App.Helper.Interop
{
    public static class LocalStorage
    {
        public static Task SetItem(string key, string value)
        {
            return JSRuntime.Current.InvokeAsync<object>("localStorageInterop.setItem",  key, value);
        }

        public static Task<string> GetItem(string key)
        {
            return JSRuntime.Current.InvokeAsync<string>("localStorageInterop.getItem", key);
        }

        public static Task RemoveItem(string key)
        {
            return JSRuntime.Current.InvokeAsync<object>("localStorageInterop.removeItem", key);
        }

        public static Task Clear()
        {
            return JSRuntime.Current.InvokeAsync<object>("localStorageInterop.clear");
        }

        public static async Task<Dictionary<string, string>> GetAllItems()
        {
            var items = await JSRuntime.Current.InvokeAsync<string[][]>("localStorageInterop.getAllItems");
            return items.ToDictionary(x => x[0], x => x[1]);
        }
    }
}

File: src/Klemm.One/Klemm.One.Server/wwwroot/js/interop/localStorage.js
window.localStorageInterop = {
	setItem: function (key, value) {
		localStorage.setItem(key, value);
	},
	getItem: function (key) {
		return localStorage.getItem(key);
	},
	removeItem: function (key) {
		localStorage.removeItem(key);
	},
	clear: function () {
		localStorage.clear();
	},
	getAllItems: function () {
		const archive = [];
		const keys = Object.keys(localStorage);

		keys.forEach(key => {
			archive.push([key, localStorage.getItem(key)]);
		});

		return archive;
	}
};